Add support for async requests

edgar-far opened the pull request:
Related to issue: https://github.com/Subito-it/SBTUITestTunnel/issues/232

tcamin reviewed (commented):
Hi, thanks for the contribution. Could you add a test that replicates the issue with the async request. It would help to analyze the changes more in detail.
In particular I'm unsure about the changes in the `canonicalRequestForRequest` since we're now handling httpBody and httpBodyStream in different locations.

edgar-far commented:
> Hi, thanks for the contribution. Could you add a test that replicates the issue with the async request. It would help to analyze the changes more in detail. In particular I'm unsure about the changes in the `canonicalRequestForRequest` since we're now handling httpBody and httpBodyStream in different locations.

@tcamin I've updated the PR and added the test `testMonitorCapturesRequestAndResponseBodyFromAsyncRequests`, which replicates the failure in master but passes on this branch, please take a look.

tcamin commented:
Thanks, I'll take a look at this as soon as I can.

tcamin commented:
Apologies for the delay in reviewing this. To speed up the merge, I did a bit of tidy-up in [PR #238](https://github.com/Subito-it/SBTUITestTunnel/pull/238), hope you don't mind 🙂. Thanks again for the report and the contribution!
